consensus/hotstuff/timeoutcollector/timeout_cache_test.go

```diff
@@ -0,0 +1,135 @@
+package timeoutcollector
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/consensus/hotstuff/helper"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestTimeoutObjectsCache_View tests that View returns same value that was set by constructor
+func TestTimeoutObjectsCache_View(t *testing.T) {
+	view := uint64(100)
+	cache := NewTimeoutObjectsCache(view)
+	require.Equal(t, view, cache.View())
+}
+
+// TestTimeoutObjectsCache_AddTimeoutObjectRepeatedTimeout tests that AddTimeoutObject skips duplicated timeouts
+func TestTimeoutObjectsCache_AddTimeoutObjectRepeatedTimeout(t *testing.T) {
+	t.Parallel()
+
+	view := uint64(100)
+	cache := NewTimeoutObjectsCache(view)
+	timeout := helper.TimeoutObjectFixture(helper.WithTimeoutObjectView(view))
+
+	require.NoError(t, cache.AddTimeoutObject(timeout))
+	err := cache.AddTimeoutObject(timeout)
+	require.ErrorIs(t, err, RepeatedTimeoutErr)
+}
+
+// TestTimeoutObjectsCache_AddTimeoutObjectIncompatibleView tests that adding timeout with incompatible view results in error
+func TestTimeoutObjectsCache_AddTimeoutObjectIncompatibleView(t *testing.T) {
+	t.Parallel()
+
+	view := uint64(100)
+	cache := NewTimeoutObjectsCache(view)
+	timeout := helper.TimeoutObjectFixture(helper.WithTimeoutObjectView(view + 1))
+	err := cache.AddTimeoutObject(timeout)
+	require.ErrorIs(t, err, TimeoutForIncompatibleViewError)
+}
+
+// TestTimeoutObjectsCache_GetTimeout tests that GetTimeout method
+func TestTimeoutObjectsCache_GetTimeout(t *testing.T) {
+	view := uint64(100)
+	knownTimeout := helper.TimeoutObjectFixture(helper.WithTimeoutObjectView(view))
+	doubleTimeout := helper.TimeoutObjectFixture(
+		helper.WithTimeoutObjectView(view),
+		helper.WithTimeoutObjectSignerID(knownTimeout.SignerID))
+
+	cache := NewTimeoutObjectsCache(view)
+
+	// unknown timeout
+	timeout, found := cache.GetTimeoutObject(unittest.IdentifierFixture())
+	require.Nil(t, timeout)
+	require.False(t, found)
+
+	// known timeout
+	err := cache.AddTimeoutObject(knownTimeout)
+	require.NoError(t, err)
+	timeout, found = cache.GetTimeoutObject(knownTimeout.SignerID)
+	require.Equal(t, knownTimeout, timeout)
+	require.True(t, found)
+
+	// for a signer ID with a known timeout, the cache should memorize the _first_ encountered timeout
+	err = cache.AddTimeoutObject(doubleTimeout)
+	require.True(t, model.IsDoubleTimeoutError(err))
+	timeout, found = cache.GetTimeoutObject(doubleTimeout.SignerID)
+	require.Equal(t, knownTimeout, timeout)
+	require.True(t, found)
+}
+
+// TestTimeoutObjectsCache_All tests that All returns previously added timeouts in same order
```

Comment:
```suggestion
// TestTimeoutObjectsCache_All tests that All returns previously added timeouts.
```
They're stored in a map so they wouldn't be returned in the same order as they were added currently. Do we need them to be?